File: src/main/java/jp/co/sunarch/apps/idprovider/model/oauth2/EndpointRequestFactory.java
package jp.co.sunarch.apps.idprovider.model.oauth2;

import java.util.Map;

/**
 * 各エンドポイント用にリクエストオブジェクトを作成する
 */
public interface EndpointRequestFactory<R> {

  R createRequest(Map<String, String> parameters);

}

File: src/main/kotlin/jp/co/sunarch/apps/idprovider/model/oauth2/EndpointRequestFactory.kt
package jp.co.sunarch.apps.idprovider.model.oauth2

/**
 * 各エンドポイント用にリクエストオブジェクトを作成する
 */
interface EndpointRequestFactory<R> {

  fun createRequest(parameters: Map<String, String>): R

}
